src/core/thread/mesh_forwarder.cpp

```diff
@@ -1020,8 +1017,8 @@ uint16_t MeshForwarder::PrepareDataFrame(Mac::TxFrame         &aFrame,
 
         if (aAddMeshHeader)
         {
-            macAddrs.mSource.SetShort(aMeshSource);
-            macAddrs.mDestination.SetShort(aMeshDest);
+            macAddrs.mSource.SetShort(mMeshSource);
+            macAddrs.mDestination.SetShort(mMeshDest);
```

Comment:
While I understand the intended simplification of removing the input parameters, I am not in favor of adding this change.

The `MeshForwarder` class tracks `mMeshSource`, `mMeshDest` member variables alongside `mMacAddrs` and `mAddMeshHeader`, all associated with `mSendMessage`, which tracks the current message for direct transmission.

The `PrepareDataFrame()` method is intended for general use to construct a frame for a given message and its associated parameters.

It would be inconsistent to provide `aMessage`, `aMacAddrs`, and `aAddMeshHeader` as inputs to this method but then assume that for mesh source/destination, it should utilize the member variables.